[charts] Introduce dataset to simplify plot of data from API (#9774)
mui/mui-x: docs/data/charts/bars/bars.md

{{"demo": "BasicBars.js"}}

### Using a dataset

> If your data are stored in an array of objects, you can use the `dataset` helper prop.

Reviewer: ```suggestion
If your data is stored in an array of objects, you can use the `dataset` helper prop.
```

It accepts an array of objects such as `dataset={[{x: 1, y: 32}, {x: 2, y: 41}, ...]}`.

> You can reuse those data to define the series and axis thanks to `dataKey` properties.

Reviewer: ```suggestion
You can reuse this data when defining the series and axis thanks to the `dataKey` property.
```

> You can reuse this data when defining the series and axis thanks to the `dataKey` property.

Reviewer 2: ```suggestion
You can reuse this data when defining the series and axis, thanks to the `dataKey` property.
```
